### PR TITLE
service: add aggregations for languages

### PR DESCRIPTION
**Pull request by zzacharo**
Inter dependent with https://github.com/inveniosoftware/invenio-app-rdm/pull/442 since

### CODE REVIEW

File: invenio_rdm_records/resources/serializers/ui/schema.py [hunk 1]
```diff
@@ -64,6 +64,23 @@ def apply_idx(affiliation):
...
+    field_data = attr.get("metadata").get(field_name)
+
+    if not field_data:
+        return None
+
+    localized = []
+    for item in field_data:
+        localized.append({
+            "id": item.get("id"),
+            "title": item.get("metadata").get("title").get("en")
```
Comment (ppanero) [inline invenio_rdm_records/resources/serializers/ui/schema.py:78]: Comment: This is hardcoded for now, should use current locale

File: invenio_rdm_records/resources/serializers/ui/schema.py [hunk 2]
```diff
@@ -64,6 +64,23 @@ def apply_idx(affiliation):
     }
 
 
+def localize_vocabulary_list(field_name, key, attr, obj):
+    """Localize vocabulary."""
+    field_data = attr.get("metadata").get(field_name)
```
Comment (ppanero) [inline invenio_rdm_records/resources/serializers/ui/schema.py:69]: Is it worth improving this to receive a dotted path? e.g. get`metadata.languages` as `field_name` and process it according to the dots

File: invenio_rdm_records/resources/serializers/ui/schema.py [hunk 3]
```diff
@@ -64,6 +66,23 @@ def apply_idx(affiliation):
...
+    field_data = attr.get("metadata", {}).get(field_name)
+
+    if not field_data:
+        return None
+
+    localized = []
+    for item in field_data:
+        localized.append({
+            "id": item.get("id"),
+            "title": item.get("metadata", {}).get("title", {}).get("en")
```
Comment (ppanero) [inline invenio_rdm_records/resources/serializers/ui/schema.py:80]: This is hardcoded, should be changed once agreed.

Comment (zzacharo) [reply invenio_rdm_records/resources/serializers/ui/schema.py:80]: This can use the `current_i18n.locale` as [here](https://github.com/inveniosoftware/invenio-rdm-records/pull/319/files#diff-60d2e985baca9c1e2ddd61f92355612c6707af6901d2242e8d473a726af5cf30R188) no?

File: invenio_rdm_records/services/schemas/metadata.py [hunk 4]
```diff
@@ -465,11 +465,9 @@ class Meta:
         unknown = EXCLUDE
 
     id = SanitizedUnicode(required=True)
-    # TODO: replace ".en" with UI serialization
-    title = fields.Raw(attribute="metadata.title.en", dump_only=True)
+    title = fields.Raw(attribute="metadata.title")
```
Comment (zzacharo) [inline invenio_rdm_records/services/schemas/metadata.py:468]: Out of curiosity, why did you remove the `dump_only` option?

File: invenio_rdm_records/resources/serializers/ui/schema.py (hunk 3 above)
Comment (ppanero) [reply invenio_rdm_records/resources/serializers/ui/schema.py:80]: Yes, was just waiting for the green light on the overall implementation. But yes :)

File: invenio_rdm_records/services/schemas/metadata.py (hunk 4 above)
Comment (ppanero) [reply invenio_rdm_records/services/schemas/metadata.py:468]: Mistake, adding back.
